corehq/apps/users/models_sql.py

```diff
@@ -179,6 +184,41 @@ class RoleAssignableBy(models.Model):
     )
 
 
+class UserUpdateMeta(jsonobject.JsonObject):
+    updated_via = jsonobject.StringProperty()
+
+
+class UpdateDetails(jsonobject.JsonObject):
+    meta = jsonobject.ObjectProperty(UserUpdateMeta)
+    changes = jsonobject.DictProperty()
+
+
+class HQLogEntry(models.Model):
+    """
+    HQ Adaptation of Django's LogEntry model
+    """
+    CREATE = 1
+    UPDATE = 2
+    DELETE = 3
+
+    ACTION_FLAG_CHOICES = (
+        (CREATE, _('Create')),
+        (UPDATE, _('Update')),
+        (DELETE, _('Delete')),
+    )
+    domain = models.CharField(max_length=255, db_index=True)
+    object_type = models.CharField(max_length=255, db_index=True, choices=(
+        ('CommCareUser', 'CommCareUser'),
+        ('WebUser', 'WebUser'),
+    ))
+    object_id = models.CharField(max_length=128, db_index=True)
+    by_user_id = models.CharField(max_length=128, db_index=True)
+    details = JSONField(default=dict)  # UpdateDetails
+    message = models.TextField(_('change message'), blank=True)
+    action_time = models.DateTimeField(_('action time'), auto_now_add=True, editable=False)
+    action_flag = models.PositiveSmallIntegerField(_('action flag'), choices=ACTION_FLAG_CHOICES)
```

Comment:
nit: rename these to `time` (or `date`) and `flag` respectively

Reply:
used the standard "created_at" and then just "action" for the action.